hpx/runtime/components/server/create_component.hpp

```diff
@@ -53,8 +53,16 @@ namespace hpx { namespace components { namespace server
     naming::gid_type create(
         util::unique_function_nonser<void(void*)> const& ctor)
     {
-        Component* c = (Component*)Component::heap_type::alloc(1);
-        ctor(c);
+        void * cv = Component::heap_type::alloc(1);
+        try {
+            ctor(cv);
+        }
+        catch(...)
+        {
+            Component::heap_type::free(cv, 1);
+            return naming::invalid_gid;
+        }
+        Component *c = static_cast<Component *>(cv);
```

Comment:
This function is throwing an exception if creation goes wrong. I don't think we should silently return invalid_gid in case of an error.